### PR TITLE
fix: nft contract default icon

### PR DESCRIPTION
**Pull request by lelenei**
## Description

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

**Comment by wenluomask** [comment]
1. Please also put the default icon to Web3-NFT;
Screen Shot 2022-01-07 at 8 15 24 PM

2. The size of default icon should be 36x36, currently is 32x32;
Screen Shot 2022-01-07 at 8 18 21 PM
Screen Shot 2022-01-07 at 8 18 29 PM

**Comment by lelenei** [comment]
image

**Comment by wenluomask** [comment]
Checked.

File: packages/web3-providers/src/NFTScan/index.ts [hunk 1]
```diff
@@ -72,8 +72,8 @@ export class NFTScanAPI implements NonFungibleTokenAPI.Provider {
                     ChainId.Mainnet,
                     x.nft_contract_address,
                     x.nft_platform_name,
-                    undefined,
-                    undefined,
+                    '',
```
Comment (guanbinrui) [inline packages/web3-providers/src/NFTScan/index.ts]: Why set the `baseURI` and `iconURL` to empty strings?

File: packages/mask/src/plugins/Wallet/SNSAdaptor/SelectNftContractDialog.tsx [hunk 2]
```diff
@@ -255,24 +255,20 @@ function ContractListItem(props: ContractListItemProps) {
...
     const chainId = useChainId()
+    const blockie = useBlockie(contract.contractDetailed.address)
 
     return (
         <div style={{ position: 'relative' }}>
             <ListItem className={classes.listItem} onClick={() => onSubmit(contract.contractDetailed)}>
-                {contract.contractDetailed.iconURL ? (
-                    <img className={classes.icon} src={contract.contractDetailed.iconURL} />
-                ) : null}
+                <img className={classes.icon} src={contract.contractDetailed.iconURL || blockie} />
```
Comment (hanszhoou) [inline packages/mask/src/plugins/Wallet/SNSAdaptor/SelectNftContractDialog.tsx]: For `contract.contractDetailed.iconURL === ''`, we should use blockie.

`'' || 'abc' // return 'abc'`
`'' ?? 'abc' // return ''`

**Review by Jack-Works — DISMISSED** [review]

File: packages/mask/src/plugins/Wallet/SNSAdaptor/SelectNftContractDialog.tsx (hunk 2 above)
Comment (guanbinrui) [inline packages/mask/src/plugins/Wallet/SNSAdaptor/SelectNftContractDialog.tsx]: Use avatar component from MUI and no more blockie. The avatar will fallback to an alphabet.

**Comment by wenluomask** [comment]
Checked.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**